crates/matrix-sdk/src/sliding_sync/mod.rs

```diff
@@ -100,6 +101,9 @@ pub(super) struct SlidingSyncInner {
     /// Position markers
     position: StdRwLock<SlidingSyncPositionMarkers>,
 
+    /// Past position markers.
+    past_positions: StdRwLock<RingBuffer<SlidingSyncPositionMarkers>>,
```

Comment:
Any reason to store the `SlidingSyncPositionMarkers` overall, rather than just the plain `pos` as `Option<String>`? `delta_token` isn't returned by the server (or it might even be the same value, which should semantically cause an error too but won't), and we're only comparing against the `pos` fields anyways in the iteration loop below.

Reply:
I thought it was easier. We manipulate a single type. The API looks consistent. It also allows to rollback to some `pos` and `delta_token` if required, or to debug, if required. I don't have a strong opinion here.

Reply:
There's just potential to shoot ourselves in the feet later: comparing two pairs of `SlidingSyncPositionMarkers` isn't the same as independently comparing its fields against two pairs of position markers. Again, delta_token might not be even provided (or it should be a constant), but since your code only reads the `pos: Option<String>` field, it seems simpler to store only that, until there's need for more?